src/rtc.rs

```diff
@@ -476,6 +493,48 @@ impl Rtc {
         self.rtc_config = rtc_config;
     }
 
+    const RTC_CALR_MIN_OFFSET_PPM: f32 = -487.1;
+    const RTC_CALR_MAX_OFFSET_PPM: f32 = 488.5;
+    const RTC_CALR_RESOLUTION_PPM: f32 = 0.9537;
+
+    /// Calibrate the RTC's PPM offset.
+    pub fn set_ppm_offset(&mut self, mut offset: f32, period: RtcCalibrationCyclePeriod) {
+        if offset < Self::RTC_CALR_MIN_OFFSET_PPM {
+            offset = Self::RTC_CALR_MIN_OFFSET_PPM;
+        } else if offset > Self::RTC_CALR_MAX_OFFSET_PPM {
+            offset = Self::RTC_CALR_MAX_OFFSET_PPM;
+        }
+
+        offset = offset / Self::RTC_CALR_RESOLUTION_PPM;
+
+        self.write(false, |rtc| {
+            rtc.calr.modify(|_, mut w| unsafe {
+                match period {
+                    RtcCalibrationCyclePeriod::Period8 => {
+                        w = w.calw8().set_bit();
```

Comment:
This does not look right, it should only be `w.calw8().set_bit();`, same for the next 2 match arms.

Reply:
You mean `w.calw16().clear_bit()` also needs to be in this arm?

Reply:
That, and to not assign `w`.
Only `w.calw().set_bit();`, `w` is a mutable reference.

For the lines below you should only use the builder pattern:

```rust
w.calw8().clear_bit().calw16().set_bit();
```

Etc.